### PR TITLE
Add instructions to start the editor for custom servers

### PR DESCRIPTION
**Pull request by Janpot**
Complements https://github.com/mui/mui-toolpad/pull/2800 with instructions in a popover on the application header, to start the standalone editor.  

Screenshot 2023-11-07 at 17 22 56

### CODE REVIEW

File: packages/toolpad-app/src/runtime/PreviewHeader.tsx [hunk 1]
```diff
@@ -1,10 +1,178 @@
...
 import { useMatch } from 'react-router-dom';
+import ContentCopyIcon from '@mui/icons-material/ContentCopy';
 import { IS_CUSTOM_SERVER, PREVIEW_HEADER_HEIGHT } from './constants';
 
-export default function PreviewHeader() {
+interface CopyToClipboardButtonProps extends IconButtonProps {
+  content: string;
+}
+
+function CopyToClipboardButton({ content, onClick, ...props }: CopyToClipboardButtonProps) {
```
Comment (apedroferreira) [inline packages/toolpad-app/src/runtime/PreviewHeader.tsx:26]: Looks like this could be a useful shared component besides being just in this file?

**Review by apedroferreira — APPROVED** [review]
Looks great!

I just noticed that in the screenshot the copy icon seems a bit off-center vertically.

File: packages/toolpad-app/src/runtime/PreviewHeader.tsx (hunk 1 above)
Comment (Janpot) [reply packages/toolpad-app/src/runtime/PreviewHeader.tsx:26]: I don't want to incentivize maintainers importing from outside the runtime folder. That's why I opted for this setup. Maybe in the future.

**Comment by Janpot** [comment]
> I just noticed that in the screenshot the copy icon seems a bit off-center vertically.

🤔  It's not meant to be centered, it's meant to be in the corner

File: packages/toolpad-app/src/runtime/PreviewHeader.tsx (hunk 1 above)
Comment (apedroferreira) [reply packages/toolpad-app/src/runtime/PreviewHeader.tsx:26]: Ah got it because it's in the runtime, ok we can keep it.

**Comment by apedroferreira** [comment]
> > I just noticed that in the screenshot the copy icon seems a bit off-center vertically.
> 
> 🤔 It's not meant to be centered, it's meant to be in the corner

Ok, I personally think it would look better centered but if it's intentional feel free to keep it!

**Comment by Janpot** [comment]
> Ok, I personally think it would look better centered but if it's intentional feel free to keep it!

My main hesitation against that would be, what would it look like then for a code viewer that has multiple lines?

**Comment by apedroferreira** [comment]
> > Ok, I personally think it would look better centered but if it's intentional feel free to keep it!
> 
> My main hesitation against that would be, what would it look like then for a code viewer that has multiple lines?

Got it, yeah even in the MUI docs it's always in the corner https://mui.com/material-ui/react-autocomplete/, I didn't think of that scenario. For a single line command maybe centering still looks better aesthetically, but anyway not a big deal, I think we can have it in the corner for consistency.

**Pull request merged**